gce-containers-startup/utils/volumes.go

```diff
@@ -0,0 +1,144 @@
+package utils
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	api "github.com/konlet/types"
+)
+
+type VolumeHostPathAndMode struct {
+	// nil hostPath means no backing directory, implying tmpfs mount.
+	hostPath string
+	readOnly bool
+}
+
+type TmpFsConfiguration struct {
+	path     string
+	readOnly bool
+}
+
+type HostPathBindConfiguration struct {
+	hostPath      string
+	containerPath string
+	readOnly      bool
+}
+
+// Structure represents data that is passed to docker - host binding paths, tmpfs paths and volumes.
+type VolumeBindingConfiguration struct {
+	// Maps from container name to relevant bindings.
+	hostPathBinds []HostPathBindConfiguration
+	tmpFsBinds    []TmpFsConfiguration
+}
+
+// This is the main interface to this module.
+//
+// The function takes the API specification and:
+//  - Verifies consistency.
+//  - Creates/mounts/format all the necessary volumes.
+//  - Outputs all the binding maps, keyed by container name.
+//
+// The caller should not expect the function to be idempotent. Errors are to be considered non-retryable.
+func prepareVolumesAndGetBindings(spec api.ContainerSpecStruct) (map[string]VolumeBindingConfiguration, error) {
```

Comment:
Thanks for taking this out of the main file.
But I'm afraid this function is still way too long - please split further.

Reply:
Done.